rclcpp/test/rclcpp/wait_set_policies/test_storage_policy_common.cpp

```diff
@@ -0,0 +1,171 @@
+{
+public:
+  static void SetUpTestCase()
+  {
+    rclcpp::init(0, nullptr);
+  }
+
+  static void TearDownTestCase()
+  {
+    rclcpp::shutdown();
+  }
+
+  void SetUp()
+  {
+    node = std::make_shared<rclcpp::Node>("node", "ns");
+  }
+
+  std::shared_ptr<rclcpp::Node> node;
+};
+
+class TestWaitable : public rclcpp::Waitable
+{
+public:
+  TestWaitable()
+  : is_ready_(false), add_to_wait_set_(false) {}
+  bool add_to_wait_set(rcl_wait_set_t *) override {return add_to_wait_set_;}
+
+  bool is_ready(rcl_wait_set_t *) override {return is_ready_;}
+
+  void execute() override {}
+
+  void set_is_ready(bool value) {is_ready_ = value;}
+
+  void set_add_to_wait_set(bool value) {add_to_wait_set_ = value;}
+
+private:
+  bool is_ready_;
+  bool add_to_wait_set_;
+};
+
+TEST_F(TestStoragePolicyCommon, rcl_wait_set_fini_error) {
+  auto wait_set = std::make_shared<rclcpp::WaitSet>();
+  auto mock = mocking_utils::inject_on_return(
+    "lib:rclcpp", rcl_wait_set_fini, RCL_RET_ERROR);
+  EXPECT_NO_THROW(wait_set.reset());
+}
+
+TEST_F(TestStoragePolicyCommon, rcl_wait_set_resize_error) {
+  rclcpp::WaitSet wait_set;
+
+  auto subscription = node->create_subscription<test_msgs::msg::Empty>(
+    "topic", 10, [](test_msgs::msg::Empty::SharedPtr) {});
+  rclcpp::SubscriptionWaitSetMask mask{true, true, true};
+
+  auto mock = mocking_utils::patch_and_return(
+    "lib:rclcpp", rcl_wait_set_resize, RCL_RET_ERROR);
+  wait_set.add_subscription(subscription, mask);
+  EXPECT_THROW(
+    wait_set.wait(),
+    rclcpp::exceptions::RCLError);
+}
+
+TEST_F(TestStoragePolicyCommon, rcl_wait_set_clear_error) {
+  rclcpp::WaitSet wait_set;
+
+  auto mock = mocking_utils::patch_and_return(
+    "lib:rclcpp", rcl_wait_set_clear, RCL_RET_ERROR);
+  EXPECT_THROW(
+    wait_set.wait(),
+    rclcpp::exceptions::RCLError);
+}
+
+TEST_F(TestStoragePolicyCommon, rcl_wait_set_add_subscription_error) {
+  rclcpp::WaitSet wait_set;
+  auto subscription = node->create_subscription<test_msgs::msg::Empty>(
+    "topic", 10, [](test_msgs::msg::Empty::SharedPtr) {});
+  rclcpp::SubscriptionWaitSetMask mask{true, true, true};
+
+  auto mock = mocking_utils::patch_and_return(
+    "lib:rclcpp", rcl_wait_set_add_subscription, RCL_RET_ERROR);
+  wait_set.add_subscription(subscription, mask);
+  EXPECT_THROW(
+    wait_set.wait(),
+    rclcpp::exceptions::RCLError);
+}
+TEST_F(TestStoragePolicyCommon, rcl_wait_set_add_guard_condition_error) {
+  rclcpp::WaitSet wait_set;
+  auto guard_condition = std::make_shared<rclcpp::GuardCondition>();
+  auto mock = mocking_utils::patch_and_return(
+    "lib:rclcpp", rcl_wait_set_add_guard_condition, RCL_RET_ERROR);
+  wait_set.add_guard_condition(guard_condition);
+  EXPECT_THROW(
+    wait_set.wait(),
+    rclcpp::exceptions::RCLError);
+}
+
+TEST_F(TestStoragePolicyCommon, rcl_wait_set_add_timer_error) {
+  rclcpp::WaitSet wait_set;
+  // This is long, so it can stick around and be removed
```

Comment:
I think this comment is not necessary here

Reply:
Removed